appserver/ejb/ejb-http-remoting/admin/src/main/java/fish/payara/ejb/http/admin/EjbInvokerService.java

```diff
@@ -0,0 +1,121 @@
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.ejb.http.admin;
+
+import com.sun.enterprise.config.serverbeans.SecurityService;
+import com.sun.enterprise.deployment.Application;
+import com.sun.enterprise.deployment.WebBundleDescriptor;
+import static com.sun.enterprise.deployment.runtime.web.SunWebApp.HTTPSERVLET_SECURITY_PROVIDER;
+import com.sun.enterprise.deployment.web.LoginConfiguration;
+import com.sun.enterprise.util.StringUtils;
+import static fish.payara.ejb.http.admin.Constants.EJB_INVOKER_APP;
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import static javax.servlet.http.HttpServletRequest.FORM_AUTH;
+import org.glassfish.api.StartupRunLevel;
+import org.glassfish.api.deployment.DeploymentContext;
+import org.glassfish.api.event.EventListener;
+import org.glassfish.api.event.Events;
+import org.glassfish.hk2.runlevel.RunLevel;
+import org.glassfish.internal.api.Globals;
+import org.glassfish.internal.deployment.Deployment;
+import org.glassfish.web.deployment.runtime.SunWebAppImpl;
+import org.jvnet.hk2.annotations.Service;
+/**
+ *
+ * @author Gaurav Gupta
+ */
+@Service(name = "ejb-invoker-service")
+@RunLevel(StartupRunLevel.VAL)
+public class EjbInvokerService implements EventListener {
+    
+    @Inject
+    private Events events;
+    
+    @Inject
+    private SecurityService securityService;
+
+    @Inject
+    private EjbInvokerConfiguration config;
+
+    @PostConstruct
+    public void init() {
+        if (events == null) {
+            events = Globals.getDefaultBaseServiceLocator().getService(Events.class);
+        }
+        events.register(this);
+    }
+
+    @Override
+    public void event(EventListener.Event event) {
+        if (event.is(Deployment.APPLICATION_PREPARED)) {
+            DeploymentContext context = (DeploymentContext) event.hook();
+            Application app = context.getModuleMetaData(Application.class);
+            if (EJB_INVOKER_APP.equals(app.getAppName())
```

Comment:
we should consider giving the app less generic name, so it wouldn't accidentally clash with user application, applying unexpected configuration to it, especially if user is completely unaware of EJB Http Adapter.
Probably prefixing it with an underscore would do.

Reply:
Renamed application to `__ejb-invoker`